### PR TITLE
Minor change to fix the incorrect response truncation

### PR DESCRIPTION
**Pull request by amankhandelia**
Following code changes fixes the bug identified in #3985

### CODE REVIEW

File: ludwig/features/text_feature.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ def get_decoded_targets_and_predictions(
     """Returns the decoded targets and predictions, accounting for IGNORE_INDEX_TOKEN_ID."""
     sanitized_targets = torch.where(targets != IGNORE_INDEX_TOKEN_ID, targets, tokenizer.pad_token_id)
     sanitized_predictions = torch.where(
-        predictions[PREDICTIONS] != IGNORE_INDEX_TOKEN_ID,
+        targets != IGNORE_INDEX_TOKEN_ID,
```
Comment (alexsherstinsky) [inline ludwig/features/text_feature.py:71]: @amankhandelia Thank you so much for finding the cause and proposing a fix.  Would it be possible for you as part of this pull request to include a test for the correction you made?  That would be really helpful, and hopefully would not take too much of your time.  Thanks again!

**Review by alexsherstinsky — CHANGES_REQUESTED** [review]
Looks great -- just would hope to also include a test.  Thank you!

**Comment by amankhandelia** [comment]
@alexsherstinsky, have added a regression test case for the fix. Please review the same.

File: tests/ludwig/features/test_text_feature.py [hunk 2]
```diff
@@ -124,5 +134,9 @@ def test_get_decoded_targets_and_predictions():
         decoded_predictions,
     ) = text_feature.get_decoded_targets_and_predictions(targets, predictions, tokenizer)
 
-    assert decoded_targets == [" first she 18 yearman our<s>"]
-    assert decoded_predictions == [" first she 18 yearman our<s>"]
+    assert isinstance(decoded_targets, list)
+    assert isinstance(decoded_predictions, list)
+    assert all(isinstance(x, str) for x in decoded_targets)
+    assert all(isinstance(x, str) for x in decoded_predictions)
+    assert decoded_targets == decoded_predictions
```
Comment (alexsherstinsky) [inline tests/ludwig/features/test_text_feature.py:141]: @amankhandelia One small question: would it work if we said `assert decoded_targets == decoded_predictions == decoded_texts` for concision?  Thank you.

Comment (amankhandelia) [reply tests/ludwig/features/test_text_feature.py]: yes, that would work, I have made the change to that effect

Comment (alexsherstinsky) [reply tests/ludwig/features/test_text_feature.py]: @amankhandelia Thank you very much for making the change.  At this point, I want to consult with a teammate (just to be 100% sure) -- so I will get back to you within 12 -- 13 hours from now.  Thanks for your patience!

**Review by alexsherstinsky — APPROVED** [review]
LGTM

**Comment by alexsherstinsky** [comment]
@amankhandelia thank you very much -- approved; will be in the release coming later today!

**Pull request merged**